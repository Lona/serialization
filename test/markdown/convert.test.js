const { convertDocument } = require('../../lib/index')

describe('mdx <-> json', () => {
  describe('root', () => {
    const mdx = `# heading

a`

    const json = {
      children: [
        {
          type: 'heading',
          data: {
            depth: 1,
            children: [
              {
                type: 'text',
                data: {
                  value: 'heading',
                  children: [],
                },
              },
            ],
          },
        },
        {
          type: 'paragraph',
          data: {
            children: [
              {
                type: 'text',
                data: {
                  value: 'a',
                  children: [],
                },
              },
            ],
          },
        },
      ],
    }

    test('mdx -> json', () => {
      const convertedToJson = JSON.parse(
        convertDocument(mdx, 'json', { sourceFormat: 'mdx' })
      )

      expect(convertedToJson).toStrictEqual(json)
    })

    test('json -> mdx', () => {
      const convertedToMdx = convertDocument(JSON.stringify(json), 'mdx', {
        sourceFormat: 'json',
      })

      expect(convertedToMdx).toBe(mdx)
    })
  })

  describe('image', () => {
    const mdx = `![alt](url)`

    const json = {
      children: [
        {
          type: 'paragraph',
          data: {
            children: [
              {
                type: 'image',
                data: {
                  alt: 'alt',
                  url: 'url',
                  title: null,
                  children: [],
                },
              },
            ],
          },
        },
      ],
    }

    test('mdx -> json', () => {
      const convertedToJson = JSON.parse(
        convertDocument(mdx, 'json', { sourceFormat: 'mdx' })
      )

      expect(convertedToJson).toStrictEqual(json)
    })

    test('json -> mdx', () => {
      const convertedToMdx = convertDocument(JSON.stringify(json), 'mdx', {
        sourceFormat: 'json',
      })

      expect(convertedToMdx).toBe(mdx)
    })
  })
})
