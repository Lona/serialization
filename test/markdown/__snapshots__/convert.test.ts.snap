// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mdx <-> json tokens 1`] = `
"{
  \\"children\\": [
    {
      \\"data\\": {
        \\"children\\": [
          {
            \\"data\\": {
              \\"value\\": \\"hello\\"
            },
            \\"type\\": \\"text\\"
          }
        ]
      },
      \\"type\\": \\"paragraph\\"
    },
    {
      \\"data\\": {
        \\"lang\\": \\"tokens\\",
        \\"meta\\": null,
        \\"parsed\\": {
          \\"data\\": {
            \\"declarations\\": [
              {
                \\"data\\": {
                  \\"annotation\\": {
                    \\"data\\": {
                      \\"genericArguments\\": [
                      ],
                      \\"id\\": \\"0\\",
                      \\"identifier\\": {
                        \\"id\\": \\"0\\",
                        \\"isPlaceholder\\": false,
                        \\"string\\": \\"Color\\"
                      }
                    },
                    \\"type\\": \\"typeIdentifier\\"
                  },
                  \\"id\\": \\"0\\",
                  \\"initializer\\": {
                    \\"data\\": {
                      \\"id\\": \\"0\\",
                      \\"literal\\": {
                        \\"data\\": {
                          \\"id\\": \\"0\\",
                          \\"value\\": \\"red\\"
                        },
                        \\"type\\": \\"color\\"
                      }
                    },
                    \\"type\\": \\"literalExpression\\"
                  },
                  \\"name\\": {
                    \\"id\\": \\"0\\",
                    \\"name\\": \\"test\\"
                  }
                },
                \\"type\\": \\"variable\\"
              },
              {
                \\"data\\": {
                  \\"id\\": \\"0\\"
                },
                \\"type\\": \\"placeholder\\"
              }
            ],
            \\"id\\": \\"0\\"
          },
          \\"type\\": \\"topLevelDeclarations\\"
        },
        \\"value\\": \\"let test: Color = #color(css: \\\\\\"red\\\\\\")\\"
      },
      \\"type\\": \\"code\\"
    }
  ]
}"
`;
